### PR TITLE
Issue #94 is partially solved + refactoring of backend factory

### PR DESCRIPTION
**Pull request by sergehog**

### CODE REVIEW

File: hallux/backends/factory.py [hunk 1]
```diff
@@ -36,6 +44,40 @@ def init_backend(argv: list[str], config: list[dict] | None, config_path: Path)
...
+            is_extra_arg_present = len(argv) > max_args
+            is_last_arg_not_option = not argv[-1].startswith("--")
+
+            if is_extra_arg_present and is_last_arg_not_option:
+                path_value = argv[-1]
+                if path_value:
+                    return model_value
+                else:
+                    raise SystemError("Missing last path argument. Please provide a valid path")
+                    # logger.error("Missing last path argument. Please provide a valid path.")
```
Comment (sergehog) [inline hallux/backends/factory.py:64]: commented code

File: hallux/backends/factory.py [hunk 2]
```diff
@@ -36,6 +44,40 @@ def init_backend(argv: list[str], config: list[dict] | None, config_path: Path)
...
+                    # return None
+            else:
+                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
+                # logger.error("Missing or invalid last path argument. Please provide a valid path.")
+                # return None
```
Comment (HalluxAIDev) [inline hallux/backends/factory.py:69]: Remove this commented out code.
```suggestion
            else:
                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
```

File: hallux/main.py [hunk 3]
```diff
@@ -149,34 +149,6 @@ def get_version():
...
-            return None
-    else:
-        logger.error(
-            "The '--model' argument must be followed by a valid model name, like '--model=gpt4o'.\n"
-            "More details on model options: https://hallux.dev/docs/user-guide/backends"
-        )
-        return None
-
-
 def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```
Comment (HalluxAIDev) [inline hallux/main.py:152]: Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```

File: hallux/tools/factory.py [hunk 4]
```diff
@@ -12,7 +12,7 @@
 from ..tools.sonarqube.solver import Sonar_IssueSolver
 
 
-class ProcessorFactory:
+class ToolsFactory:
     @staticmethod
     def init_solvers(
```
Comment (HalluxAIDev) [inline hallux/tools/factory.py:17]: Refactor this function to reduce its Cognitive Complexity from 28 to the 15 allowed.
```suggestion
    def init_solvers(
```

File: hallux/backends/factory.py (hunk 2 above)
Comment (sergehog) [inline hallux/backends/factory.py:69]: Remove this commented out code.
```suggestion
            has_delimeter = "=" in argv[model_index]
            max_args = model_index + (1 if has_delimeter else 2)
            is_extra_arg_present = len(argv) > max_args
            is_last_arg_not_option = not argv[-1].startswith("--")

            if is_extra_arg_present and is_last_arg_not_option:
                path_value = argv[-1]
                if path_value:
                    return model_value
                else:
                    raise SystemError("Missing last path argument. Please provide a valid path")
            else:
                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
```

File: hallux/main.py (hunk 3 above)
Comment (sergehog) [inline hallux/main.py:152]: Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```

File: hallux/backends/factory.py (hunk 2 above)
Comment (sergehog) [inline hallux/backends/factory.py:69]: Remove this commented out code.
```suggestion
            has_delimeter = "=" in argv[model_index]
            max_args = model_index + (1 if has_delimeter else 2)
            is_extra_arg_present = len(argv) > max_args
            is_last_arg_not_option = not argv[-1].startswith("--")

            if is_extra_arg_present and is_last_arg_not_option:
                path_value = argv[-1]
                if path_value:
                    return model_value
                else:
                    raise SystemError("Missing last path argument. Please provide a valid path")
            else:
                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
```

File: hallux/main.py (hunk 3 above)
Comment (sergehog) [inline hallux/main.py:152]: Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```

File: hallux/backends/factory.py (hunk 2 above)
Comment (sergehog) [inline hallux/backends/factory.py:69]: Remove this commented out code.
```suggestion
            has_delimeter = "=" in argv[model_index]
            max_args = model_index + (1 if has_delimeter else 2)
            is_extra_arg_present = len(argv) > max_args
            is_last_arg_not_option = not argv[-1].startswith("--")

            if is_extra_arg_present and is_last_arg_not_option:
                path_value = argv[-1]
                if path_value:
                    return model_value
                else:
                    raise SystemError("Missing last path argument. Please provide a valid path")
            else:
                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
```

Comment (sergehog) [inline hallux/backends/factory.py:69]: Remove this commented out code.
```suggestion
            else:
                raise SystemError("Missing or invalid last path argument. Please provide a valid path.")
```

File: hallux/main.py (hunk 3 above)
Comment (sergehog) [inline hallux/main.py:152]: Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```

Comment (HalluxAIDev) [inline hallux/main.py:152]: Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int: 
```

File: hallux/tools/factory.py (hunk 4 above)
Comment (HalluxAIDev) [inline hallux/tools/factory.py:17]: Refactor this function to reduce its Cognitive Complexity from 28 to the 15 allowed.
```suggestion
    def init_solvers(
```

**Pull request merged**